lib/terraform/runner/credential.rb

```diff
@@ -0,0 +1,31 @@
+module Terraform
+  class Runner
+    class Credential
+      attr_reader :auth
+
+      def self.new(authentication_id)
```

Comment:
Suggest we use `find` so it looks like active record find where you pass an id.  It doesn't matter if it creates a new object or find an existing one, we only care that it returns an instance of the proper subclass object.

Reply:
Again used the same pattern, as in ansible implementation, https://github.com/ManageIQ/manageiq/blob/master/lib/ansible/runner/credential.rb#L7

Yes it uses `find` but I suppose, `new` also makes sense, as creates a new instance of `Terraform::Runner::Credential`

Reply:
Yeah, at second thought, this is like a builder/factory method.  It's fine but generally `new` creates instances of the class whereas a builder/factory could build new objects based on the inputs and they don't need to be the same class.  We can keep it if ansible did it this way, but I try to avoid overriding new.